packages/expo-cli/src/commands/build/AndroidBuilder.js

```diff
@@ -212,11 +212,20 @@ export default class AndroidBuilder extends BaseBuilder {
   async validateProject(options) {
     const publicUrl = options.publicUrl;
     const {
-      args: { sdkVersion },
+      args: { sdkVersion, androidPackage },
     } = publicUrl
-      ? await Exp.getThirdPartyInfoAsync(publicUrl)
-      : await Exp.getPublishInfoAsync(this.projectDir);
+      ? await Exp.getThirdPartyInfoAsync(publicUrl, 'android')
+      : await Exp.getPublishInfoAsync(this.projectDir, 'android');
```

Comment:
`Exp.getThirdPartyInfoAsync` also needs to be updated to return the `androidPackage` field.